arm9/src/unk_0200BB14.c

```diff
@@ -0,0 +1,1280 @@
+        param1->unk0c[0],
+        param1->unk0c[1],
+        param1->unk0c[2],
+        param1->unk0c[3],
+        param1->unk0c[4],
+        param1->unk0c[5]);
+    FreeToHeap(r6);
+
+    return 1;
+}
+
+THUMB_FUNC u32 FUN_0200BE38(u32 *param0, u32 *param1, struct UnkStruct_0200BB14_3 *param2)
+{
+    return FUN_0200BE74(param0,
+        param1,
+        param2->unk00,
+        param2->unk04,
+        param2->unk06,
+        param2->unk04,
+        param2->unk0a,
+        param2->unk0c,
+        param2->unk10,
+        param2->unk14,
+        param2->unk18,
+        param2->unk1c,
+        param2->unk20,
+        param2->unk24);
+}
+
+THUMB_FUNC u32 FUN_0200BE74(u32 *param0,
+    u32 *param1,
+    u32 param2,
+    s16 param3,
+    s16 param4,
+    s16 param5,
+    u16 param6,
+    u32 param7,
+    u32 param8,
+    u32 param9,
+    u32 param10,
+    u32 param11,
+    u32 param12,
+    u32 param13)
+{
+#pragma unused(param10)
+#pragma unused(param11)
+#pragma unused(param12)
+#pragma unused(param13)
+
+    struct UnkStruct_0200BB14_4 st0;
+    st0.unk00 = param1[0];
+    st0.unk04 = &(*(struct UnkStruct_0200BB14_sub **)(param1[1]))[param2];
+
+    float r0;
+    if (param3 > 0)
+    {
+        r0 = ((float)(param3 << 0xc) + (float)0.5);
+    }
+    else
+    {
+        r0 = ((float)(param3 << 0xc) - (float)0.5);
+    }
+    st0.unk08 = (s32)r0;
+
+    if (param4 > 0)
+    {
+        r0 = ((float)(param4 << 0xc) + (float)0.5);
+    }
+    else
+    {
+        r0 = ((float)(param4 << 0xc) - (float)0.5);
+    }
+    st0.unk0c = (s32)r0;
+
+    if (param5 > 0)
+    {
+        r0 = ((float)(param5 << 0xc) + (float)0.5);
+    }
+    else
+    {
+        r0 = ((float)(param5 << 0xc) - (float)0.5);
+    }
+    st0.unk10 = (s32)r0;
+
+    if (param9 == 2)
+    {
+        st0.unk0c += 0xC0000;
+    }
+
+    st0.unk14 = 0x1000;
+    st0.unk18 = 0x1000;
+    st0.unk1c = 0x1000;
+    st0.unk20 = 0;
+    st0.unk24 = param7;
+    st0.unk28 = param9;
+    st0.unk2c = param0[0];
+
+    u32 r4 = FUN_0201FE94(&st0);
+    if (r4 != 0)
+    {
+        FUN_02020130(r4, param6);
+
+        FUN_02020248(r4, param8);
+    }
+
+    return r4;
+}
+
+THUMB_FUNC u32 FUN_0200BF60(struct UnkStruct_0200BB14_1 *param0, struct UnkStruct_0200BB14_2 *param1, u32 *param2)
+{
+    struct UnkStruct_0200BB14_1 *st0 = param0;
+    int i;
+    int st4 = 6;
+
+    if (st0 == NULL || param1 == NULL)
+    {
+        return 0;
+    }
+
+    if (param2[4] == 0 || param2[5] == 0)
+    {
+        st4 = 4;
+    }
+
+    param1->unk54 = st4;
+
+    for (i = 0; i < st4; i++)
+    {
+        param1->unk0c[i] = FUN_02008DEC(param2[i], i, st0->unk000);
+    }
+
+    for (i = 0; i < st4; i++)
+    {
+        if (param2[i] != 0)
+        {
+            param1->unk24[i] = FUN_02009424(param2[i], st0->unk000);
+            param1->unk3c[i] = 0;
+
+            for (int j = 0; j < (int)((u32 *)param1->unk24[i])[1]; j++)
+            {
+                ((u32 *)(((u32 *)param1->unk24[i])[0]))[j] = 0;
```

Comment:
this is almost certainly a fakematch

Reply:
I cleaned it up a bit, but yes this most likely requires a struct. I'm just hesitant to do that now because I already created a lot of them and I think we're missing some context.